core/include/kernel/dt.h

```diff
@@ -66,11 +67,35 @@ enum dt_driver_type {
 	DT_DRIVER_CLK,
 };
 
+/*
+ * dt_driver_probe_func - Callback probe function for a driver.
+ *
+ * @fdt; FDT base address
+ * @nodeoffset: Offset of the node in the FDT
+ * @compat_data: Data registered for the compatible that probed the device
+ *
+ * Return TEE_SUCCESS on successful probe,
+ *	TEE_ERROR_ITEM_NOT_FOUND when no driver matched node's compatible string
+ *	Any other TEE_ERROR_* compliant code.
+ */
+typedef TEE_Result (*dt_driver_probe_func)(const void *fdt, int nodeoffset,
+					   const void *compat_data);
+
+/*
+ * Driver instance registered to be probed on compatible node found in the DT.
+ *
+ * @name: Driver name
+ * @type: Drive type
+ * @match_table: Compatible matching identifiers, null terminated
+ * @driver: Driver private reference or NULL
+ * @probe: Probe callback (see dt_driver_probe_func) or NULL
+ */
 struct dt_driver {
 	const char *name;
 	enum dt_driver_type type;
 	const struct dt_device_match *match_table; /* null-terminated */
 	const void *driver;
+	TEE_Result (*probe)(const void *fdt, int node, const void *compat_data);
```

Comment:
Why not use the typedef?

Reply:
true, thanks. I'll fix (actaully it's fixed by a later commit in the series).